tests/conftest.py

```diff
@@ -28,6 +28,18 @@ def app(request):
     return _app
 
 
+@pytest.fixture(scope="function")
+def single_use_app():
+    """
+    A function-scoped app fixture. This should only be used for testing the static site building process, as that
+    process requires an app which has not yet handled any requests. This is the case for all management commands, which
```

Comment:
"an app which has not yet handled any requests"... this is weird.  I'm not saying it's wrong, but news to me. I know Heroku spins up a short-lived app to run the management command, but where does the "has not yet handled any requests" get enforced? (Just out of interest)

Reply:
Part of the `do_it` method does this (note: added by me, recently):

```python
@application.context_processor
        def enforce_static_mode():
            return dict(static_mode=True)
```

You can only create context processors for an application that hasn't yet handled requests, or the flask app gets upset. It's our code, rather than Heroku or anything else, that needs the app running a management command to not have handled requests yet. This is something I've really only recognised tonight - it wasn't a limitation I intentionally put in place.

Reply:
Aha, OK that makes sense.  Yeah I wouldn't have spotted that at all, it's only when these things come back to bite you!